pandas/core/generic.py

```diff
@@ -9094,9 +9093,12 @@ def mask(
         cond = com.apply_if_callable(cond, self)
 
         # see gh-21891
-        if not hasattr(cond, "__invert__"):
+        if not hasattr(cond, "__array__"):
             cond = np.array(cond)
 
+        cond[isna(cond)] = False
```

Comment:
why this?

Reply:
I want to ensure we fillna with False before taking the inverse and _then_ filling with False (which is the same as filling with True here, which we don't want). If values are missing here we want them to be treated as False so that we don't apply the mask.